azure-image/99_crate.cfg

```diff
@@ -0,0 +1,32 @@
+#cloud-config
+# vim: syntax=yaml
+# /etc/cloud/cloud.cfg.d/99_crate.cfg
+
+package_upgrade: true
+packages:
+ - crate
+runcmd:
+ - 'mkdir -pv /mnt/crate'
+ - 'chown crate:crate /mnt/crate'
+ - "export MEM_KB=$(cat /proc/meminfo | grep MemTotal | awk '{print $2}')"
+ - 'echo "" >> /etc/default/crate'
+ - 'echo "The section below was automatically generated by the cloud-init script." >> /etc/default/crate'
+ - 'echo "CRATE_HEAP_SIZE=$(($MEM_KB/1024/2))m" >> /etc/default/crate'
+ - 'echo "" >> /etc/crate/crate.yml'
+ - 'echo "" >> /etc/crate/crate.yml'
+ - 'echo "########## Azure VM Image Default Settings ##########" >> /etc/crate/crate.yml'
+ - 'echo "# The section below was automatically generated by the cloud-init script." >> /etc/crate/crate.yml'
+ - 'echo "# disable swapping"'
+ - 'echo "bootstrap.mlockall: true" >> /etc/crate/crate.yml'
+ - 'echo "# use hostname as node name"'
+ - 'echo "node.name: $(hostname)" >> /etc/crate/crate.yml'
+ - 'echo "# use private IP interface" >> /etc/crate/crate.yml'
+ - 'echo "network.host: _eth0_" >> /etc/crate/crate.yml'
+ - 'echo "# use temporary ssd disks for data" >> /etc/crate/crate.yml'
+ - 'echo "path.data: /mnt/crate" >> /etc/crate/crate.yml'
+ - 'echo "# enable unicast discovery" >> /etc/crate/crate.yml'
+ - 'echo "discovery.zen.ping.multicast.enabled: false" >> /etc/crate/crate.yml'
+ - 'echo "# discovery.zen.ping.unicast.hosts: $(hostname):4300" >> /etc/crate/crate.yml'
```

Comment:
because it is just an example. this setting needs to be provided by the user itself. it needs to be documentend in the "how to" of the VM image.
however, it could be removed here